### ISSUE TITLE
FailsafeLogger StackOverflowError 还没改么？都提了这么久了。

### ISSUE THREAD

**Issue by limingwei** [apache/incubator-dubbo#2095]
- [ ] I have searched the [issues](https://github.com/apache/incubator-dubbo/issues) of this repository and believe that this is not a duplicate.
- [ ] I have checked the [FAQ](https://github.com/apache/incubator-dubbo/wiki/FAQ) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.6.2
* Operating System version: centos
* Java version: 1.8.0_144

### Step to reproduce this issue

1. ip获取工具错误导致 StackOverflowError

### Expected Result

不报错

### Actual Result

报错，不能启动

If there is an exception, please attach the exception trace:

```
 187 "main" #1 prio=5 os_prio=0 tid=0x00007f817000b000 nid=0x48ce in Object.wait() [0x00007f817645b000]
 188    java.lang.Thread.State: WAITING (on object monitor)
 189         at java.lang.Object.wait(Native Method)
 190         at java.lang.Object.wait(Object.java:502)
 191         at java.net.InetAddress.checkLookupTable(InetAddress.java:1393)
 192         - locked <0x00000005c07e62a8> (a java.util.HashMap)
 193         at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1310)
 194         at java.net.InetAddress.getLocalHost(InetAddress.java:1500)
 195         - locked <0x00000005c07e6298> (a java.lang.Object)
 196         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress0(NetUtils.java:188)
 197         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress(NetUtils.java:180)
 198         at com.alibaba.dubbo.common.utils.NetUtils.getLocalHost(NetUtils.java:146)
 199         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.appendContextMessage(FailsafeLogger.java:40)
 200         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.warn(FailsafeLogger.java:110)
 201         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress0(NetUtils.java:220)
 202         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress(NetUtils.java:180)
 203         at com.alibaba.dubbo.common.utils.NetUtils.getLocalHost(NetUtils.java:146)
 204         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.appendContextMessage(FailsafeLogger.java:40)
 205         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.error(FailsafeLogger.java:134)
 206         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress0(NetUtils.java:222)
 207         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress(NetUtils.java:180)
 208         at com.alibaba.dubbo.common.utils.NetUtils.getLocalHost(NetUtils.java:146)
 209         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.appendContextMessage(FailsafeLogger.java:40)
 210         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.warn(FailsafeLogger.java:110)
 211         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress0(NetUtils.java:220)
 212         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress(NetUtils.java:180)
 213         at com.alibaba.dubbo.common.utils.NetUtils.getLocalHost(NetUtils.java:146)
 214         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.appendContextMessage(FailsafeLogger.java:40)
 215         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.error(FailsafeLogger.java:134)
 216         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress0(NetUtils.java:222)
 217         at com.alibaba.dubbo.common.utils.NetUtils.getLocalAddress(NetUtils.java:180)
 218         at com.alibaba.dubbo.common.utils.NetUtils.getLocalHost(NetUtils.java:146)
 219         at com.alibaba.dubbo.common.logger.support.FailsafeLogger.appendContextMessage(FailsafeLogger.java:40)
```

**Comment by diecui1202** [comment]
hi, @limingwei This bug has been fixed, and will be release in next version, maybe 2.6.3.

https://github.com/apache/incubator-dubbo/commit/70ec90dae80cf1c1537f0dd83f8a844a905ebbdb

**Comment by diecui1202** [comment]
Feel free to reopen it.

**Comment by chickenlj** [comment]
Have been merged to 2.6.x branch
